src/documents/CoreAPI.md

```diff
@@ -0,0 +1,635 @@
+hand resolution of the next path segment to that function if there is no corresponding concrete member. If `fluid.get` navigates off the end of the available
+model material, this will not trigger an error - all successive indirections will simply return `undefined`. `fluid.get` accepts various esoteric piece of configuration
+allowing it to operate indirection rules encoded in arbitrary JSON material - these are imperfectly supported, contact us if you are interested.
+
+* `model {Any}` The model object to be indirected into. This may be `undefined`, etc. without triggering an error.
+* `path {String|Array of String}` Either a period-separated [EL path](FrameworkConcepts.md#el-paths) representing a chain of properties to be navigated, or else an array of
+path segments representing this
+* Returns: `{Any}` The result of the indirection of the path into the supplied model
+
+### fluid.getImmediate(model, path) 
+
+An extremely simplistic, high-performance variant of `fluid.get` that supports none of the same bells and whistles - it is only capable of navigating over concrete
+object properties that are actually present in the model, and the path must have already been parsed out into an array of segments. 
+It still, however, operates the same semantic as `fluid.get` for navigating over undefined values.
+
+* `model {Any}` The model object to be indirected into. This may be `undefined`, etc. without triggering an error.
+* `path {Array of String}` An array of path segments representing representing a chain of properties to be navigated
+* Returns: `{Any}` The result of the indirection of the path into the supplied model
+
+### fluid.set(model, path, newValue)
+
+Writes a new value into a model at a specified path. This is performed by directly modifying the penultimately referenced model object (the one referenced by the first `n-1` 
+path segments in `path`) by writing the value `newValue` to its property named after the final path segment. Other approaches to this task are possible 
+(see [New New Notes on the ChangeApplier](https://wiki.fluidproject.org/display/fluid/New+New+Notes+on+the+ChangeApplier#NewNewNotesontheChangeApplier-Theseveralwaysofactuallyapplyingachange)) but the
+fundamental invariant to be respected is that 
+
+```
+    fluid.get(model, fluid.set(model, path, value), path) === value
+```
+Even a functional programmer could admit that. A slight wrinkle, however, is the behaviour of the special path `""` or `[]`, attempting to write a value at the model 
+root. This API is not capable of this and will fail in this case - since it is hampered at the language level by being unable to rewrite the object reference `model` which exists
+in the scope of its caller. The [ChangeApplier](ChangeApplierAPI.md) handles this situation regularly since under its new semantics, the reference `model` is at a known
```

Comment:
This discussion of the ChangeApplier would be better phrased as an instruction to use the ChangeApplier in this case (if that's what you mean) rather than a conversational comment referencing new vs old semantics, which is irrelevant to API documentation.